ansible_rulebook/terminal.py

```diff
@@ -0,0 +1,191 @@
+            *******************************************************************
+
+        The number of asterixes varies as it is limited by the terminal window
+        (if there is one) width or 80 columnns otherwise.
+        """
+        start = re.compile(
+            r"^\*\* \d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d{6} "
+            r"\[" + banner + r"\] \*{0,}$"
+        )
+        end = re.compile(r"^\*{1,}$")
+
+        banners = []
+        current_banner = []
+        for line in content.strip().splitlines():
+            if current_banner:
+                current_banner.append(line)
+                if re.search(end, line):
+                    banners.append(os.linesep.join(current_banner))
+                    current_banner = []
+                continue
+
+            if re.search(start, line):
+                current_banner.append(line)
+
+        if current_banner:
+            raise DisplayBannerIncompleteError(
+                f"banner '{banner}' incomplete: {current_banner}"
+            )
+
+        return banners
+
+    ###########################################################################
+    def banner(
+        self,
+        banner: str = "",
+        content: typing.Any = None,
+        *,
+        pretty: bool = False,
+        level: int = logging.INFO,
+        **kwargs,
+    ) -> None:
+        banner = banner.strip()
+        if banner:
+            banner = f"[{banner}]"
+        formatted_banner = self._format_banner(banner)
+        if banner:
+            formatted_banner = f"\n{formatted_banner}"
+        self.output(
+            formatted_banner,
+            level=level,
+            file=kwargs.get("file", None),
+            flush=kwargs.get("flush", False),
+        )
+        if content is not None:
+            self.output(content, pretty=pretty, level=level, **kwargs)
+            self.banner(level=level, **kwargs)
+
+    ###########################################################################
+    def output(
+        self,
+        content: typing.Any,
+        *,
+        pretty: bool = False,
+        level: int = logging.INFO,
+        **kwargs,
+    ) -> None:
+        if level >= self.level:
+            if pretty:
+                content = pprint.pformat(content)
+            print(content, **kwargs)
+
+    ###########################################################################
+    def __init__(self, level: int = logging.INFO) -> None:
+        super().__init__()
+        self.level = level
+
+    ###########################################################################
+    def _format_banner(self, banner: str) -> str:
+        if len(banner) > 0:
+            banner = self._prefix_time(banner)
+        banner = self._rule_embed(banner)
+        return banner
+
+    ###########################################################################
+    def _prefix_time(self, content: str) -> str:
+        return f"{datetime.datetime.now().isoformat(' ')} {content}"
```

Comment:
It would be convenient to use a timezone-aware object: 
https://docs.astral.sh/ruff/rules/call-datetime-now-without-tzinfo/

Reply:
These are terminal-level outputs not log entries.  Using a timezone would produce output with timestamps inconsistent with other ansible-rulebook output terminal output.